server/controllers/TripController.js

```diff
@@ -28,6 +30,16 @@ class TripController {
       const tripDTO = trip.toObject()
       delete tripDTO.userId
       return tripDTO
+    } catch (error) {
+      throw new Error('Could not fetch trip')
+    }
+  }
+
+  async getSharedTrip(id) {
```

Comment:
Do we need this? You already edited getTrip to bypass auth if the trip is public

Reply:
Yeah cause otherwise you would have to be logged in to view a share link. This endpoint is called for fetching shared trip